thoth/adviser/resolver.py

```diff
@@ -730,7 +730,7 @@ def _prepare_initial_state(self, *, with_devel: bool) -> State:
 
             if not package_versions:
                 msg = (
-                    f"Cannot satisfy direct dependencies - direct dependencies of type {direct_dependency_name!r} "
+                    f"Cannot satisfy direct dependencies: direct dependencies of type {direct_dependency_name!r} "
```

Comment:
My question was more why this message is not linked to a specific justification? Using  advise-reporter the data analyzed if a justification has no link, it consider the message, while iterating through the justifications in the products of an advise report.

```
        product = products[0]
        justifications = product["justification"]

        for r in justifications:
                if "advisory" in r:

                    justification = r["advisory"]
                    justification_type = r["type"]

                elif "link" in r:

                    justification = r["link"]
                    justification_type = r["type"]

                else:
                    justification = r["message"]
                    justification_type = r["type"]
```

Reply:
Fair point, adjusted in 9acb0e73